bin/cqlsh.py

```diff
@@ -98,6 +99,33 @@
     ZIPLIB_DIRS = ()
 
 
+def parse_auth_provider_extended(option, opt_str, value, passed_parser):
+    splits = opt_str.split('.')
+    if len(splits) > 2:
+        raise optparse.OptionValueError("Invalid --AuthProviderExtended argument only single period should be found: %s" % opt_str)
+    if len(splits) < 2:
+        raise optparse.OptionValueError("Invalid --AuthProviderExtended must provide a param using period: %s" % opt_str)
+    key = splits[1]
+
+    if not hasattr(passed_parser.values, 'auth_provider_extended') or parser.values.auth_provider_extended is None:
+        passed_parser.values.auth_provider_extended = {}
+
+    passed_parser.values.auth_provider_extended[key] = value
```

Comment:
The name `auth_provider_extended` is little confusing in my opinion. Since it is set of properties that the pluggable AuthProvider can use, would `auth_provider_properties` or `auth_provider_config` make more sense?

Reply:
Recently changed this to auth_provider_
config.

Reply:
thanks!